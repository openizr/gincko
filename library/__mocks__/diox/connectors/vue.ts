/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/**
 * diox/connectors/vue mock.
 */
const connect = vi.fn(() => vi.fn(() => ({
  steps: [{ id: 'start' }, { id: 'end' }],
  loading: process.env.LOADING === 'true',
})));

export default connect;
