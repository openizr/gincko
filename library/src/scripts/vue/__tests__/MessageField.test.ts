/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { render } from '@testing-library/vue';
import MessageField from 'scripts/vue/MessageField.vue';

vi.mock('biuty/vue');

describe('vue/MessageField', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - default props', async () => {
    const { container } = render(MessageField, { props: { id: 'test', label: '*Test*' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - custom props', async () => {
    const { container } = render(MessageField, { props: { id: 'test', label: '*Test*', modifiers: 'strong' } });
    expect(container.firstChild).toMatchSnapshot();
  });
});
