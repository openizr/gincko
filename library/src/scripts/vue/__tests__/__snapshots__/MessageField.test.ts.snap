// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue/MessageField > renders correctly - custom props 1`] = `
<section
  class="ui-message strong"
  id="test"
>
  MARKDOWN FOR *Test*, false
</section>
`;

exports[`vue/MessageField > renders correctly - default props 1`] = `
<section
  class="ui-message"
  id="test"
>
  MARKDOWN FOR *Test*, false
</section>
`;
